pkg/cmd/roachtest/tests/multiregion_leasing.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+// Benchmarks and confirms the performance of the leasing infrastructure.
+func runAcceptanceMultiRegionBenchmarkLeasing(ctx context.Context, t test.Test, c cluster.Cluster) {
+	var durations [2]time.Duration
+	defaultOpts := install.MakeClusterSettings(install.SecureOption(true))
+	for modeIdx, sessionBasedLeasingEnabled := range []bool{true, false} {
+		func() {
+			t.Status("Setting up Session Based Leasing: ", sessionBasedLeasingEnabled)
+			// When session based leasing is disabled, force expiry based leasing.
+			if !sessionBasedLeasingEnabled {
+				c.Start(ctx, t.L(), option.DefaultStartOpts(), defaultOpts,
+					c.All())
+				options := map[string]string{
+					"sql.catalog.experimental_use_session_based_leasing": "off",
+					"sql.catalog.descriptor_lease_duration":              "30s",
+					"sql.catalog.descriptor_lease_renewal_fraction":      "6s",
+				}
+				conn := c.Conn(ctx, t.L(), c.Node(1)[0])
+				for s, v := range options {
+					if _, err := conn.Exec(fmt.Sprintf("SET CLUSTER SETTING %s='%s'", s, v)); err != nil {
+						t.Fatal(err)
+					}
+				}
+				conn.Close()
```

Comment:
Nit: to avoid leaked goroutines, it would be better to place this `defer` call right after `c.Conn`.